py-spyder ; py-spyder-devel: provide latest release and development version

reneeotten opened the pull request:
#### Description
Situation (see: https://trac.macports.org/ticket/56476):
- py-spyder provided an obsolete version (v2.3.8; released in 2015)
- py-spyder-devel provided the latest stable release (v3.2.8)

I started by changing the py-spyder-devel Portfile as follows:
- update dependencies and variants to reflect upstream's runtime/optional dependencies
- support qt4/qt5 through variants
- update patches (apply cleanly to v3.2.8; small line offsets for development version) and post-destroot phase
- add/use spyder_select

Once working as intended I then replaced the ```py-spyder```  directory with the updated ```py-spyder-devel```. Followed by some small changes in py-spyder, and updating py-spyder-devel to the latest git commit.

If needed commits can be squashed, but for ease of review keeping all commits makes most sense I think.

<!-- Note: it is best make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.13.4 17E202
Xcode 9.3.1 9E501
Python 2.7, 3.6

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?


<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

pmetzger commented:
FYI, given the sheer number of updates, I'd probably squish a bunch of these and explain all the changes in the extended part of the commit message(s).

reneeotten commented:
@pmetzger sure, that's also what I said - but it's easier to review like this. Once there is a consensus you or I can squash the commits.

eborisch commented:
@reneeotten are you interested in being the maintainer? I haven't been using Spyder for some time.

reneeotten commented:
@eborisch that's fine with me. Do you want me to co-maintain the port or do you prefer to drop your maintainership?

reneeotten commented:
@pmetzger okay - I squashed the commits to keep only three, one for each port, and I added myself as co-maintainer. I'll add a comment on Trac to alert people to this PR - just in case someone who is using Trac more than git might be interested in reviewing this.

pmetzger commented on python/py-spyder-devel/Portfile:
```diff
@@ -2,22 +2,21 @@
+PortGroup           select 1.0
 
-github.setup        spyder-ide spyder 3.2.8 v
+github.setup        spyder-ide spyder e9a2483
```
This changes the version numbering scheme so the epoch needs to be bumped.

reneeotten replied:
okay, so for the -devel port I should update that with every revision to the current date? The thing I don't understand though is that with the current version of the Portfile, ```port``` realizes that there is an update to py-spyder-devel so it doesn't seem to matter... Can I not just delete the whole thing - what am I missing here? Related to this, in the py-spyder Portfile since the ```updated to a version is numerically less than the previous version``` does not apply there, can I remove the ```epoch``` line?
Thanks!

mf2k replied:
The git tag should never be used for the version. Manually set the version to the devel version and date. For example, 3.2.8-20180522. The gimp3-devel port uses that version numbering structure. 

No need for an epoch line in that case too.

reneeotten commented on python/spyder_select/Portfile:
```diff
@@ -0,0 +1,34 @@
+
+name                spyder_select
+version             0.1
+revision            1
```
indeed, I'll change it - sorry about that...

mf2k commented on python/py-spyder-devel/Portfile:
```diff
@@ -2,22 +2,22 @@
-PortGroup           python 1.0
-
-python.versions     27 35 36
+python.versions     27 34 35 36
```
Please do not add old python versions. Let's leave that line the way it was and not add py34.

reneeotten replied:
okay, done. I think it would still be good to have some sort of MacPorts policy on which Python versions to support so that this can be done consistently...

eborisch commented:
Thanks for all your work on this; this has been on my to-do list, but I haven't had time to of late.
